common/djangoapps/student/views.py

```diff
@@ -2271,3 +2271,32 @@ def change_email_settings(request):
         track.views.server_track(request, "change-email-settings", {"receive_emails": "no", "course": course_id}, page='dashboard')
 
     return JsonResponse({"success": True})
+
+
+def _get_xseries_programs(user, user_enrolled_courses): # pylint: disable=invalid-name
+    """ Return a list of course key containing data required
+    for dashboard.
+
+    Given a user and an iterable of course keys, find all
+    the programs relevant to the user's dashboard and return them in a
+    dictionary keyed by the course_key.
+
+    Arguments:
+        user (user object): Currently logged-in User
+        course_keys (list): List of course keys in which user is enrolled
+
+    Returns:
+        Dictionary response containing programs or None
+    """
+    expected_output = get_course_programs_for_dashboard(user, user_enrolled_courses)
+    programs_data = {}
+    for course_key, course_data in expected_output.items():
+        if course_data.get('status') == 'active':
+            for course_codes_data in course_data.get('course_codes', []):
+                programs_data[course_key] = {
+                    'run_modes_count': len(course_codes_data.get('run_modes', [])),
```

Comment:
the number we want to display is the number of course codes in the program, not run modes.